File: backend/src/entity/Event.ts
import { Entity, PrimaryGeneratedColumn, Column } from "typeorm";

@Entity()
export class Event {
  @PrimaryGeneratedColumn()
  id: number = 0;

  @Column()
  title: string = "";

  @Column()
  date: string = "";

  @Column()
  time: string = "";

  @Column()
  location: string = "";

  @Column()
  ssid: string = "";
}

File: frontend/lib/event.dart
class Event {
  final int id;
  final String title;
  final String date;
  final String time;
  final String location;
  final String ssid;

  Event({
    required this.id,
    required this.title,
    required this.date,
    required this.time,
    required this.location,
    required this.ssid,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'],
      title: json['title'],
      date: json['date'],
      time: json['time'],
      location: json['location'],
      ssid: json['ssid'],
    );
  }
}
